dot/peerset/handler.go

```diff
@@ -30,67 +34,89 @@ func NewPeerSetHandler(cfg *ConfigSet) (*Handler, error) {
 	}, nil
 }
 
+func (h *Handler) setActionQueue(act action) {
+	go func(data action) {
```

Comment:
Change this to not launch a goroutine here, and instead have the caller call `go setActionQueue(act)`